File: src/cli_tools/SEF_CLI_Tool.cs

using System;
using System.Collections.Generic;
using System.IO;

class Program
{
    static void Main(string[] args)
    {
        if (args.Length == 0)
        {
            Console.WriteLine("Please provide the path to the SEF formatted exam file.");
            return;
        }

        string filePath = args[0];
        string outputPath = args.Length > 1 ? args[1] : "results.txt";

        var questions = LoadExam(filePath);
        var responses = DisplayExam(questions);
        SaveResults(responses, outputPath);

        Console.WriteLine($"Results saved to {outputPath}");
    }

    static List<string> LoadExam(string filePath)
    {
        return new List<string>(File.ReadAllLines(filePath));
    }

    static List<Tuple<string, string>> DisplayExam(List<string> questions)
    {
        Console.WriteLine("Starting the Exam:");
        var userResponses = new List<Tuple<string, string>>();
        string currentQuestion = string.Empty;
        var answers = new List<string>();

        foreach (string line in questions)
        {
            if (!line.StartsWith("-"))
            {
                if (!string.IsNullOrEmpty(currentQuestion))
                {
                    DisplayQuestion(currentQuestion, answers, userResponses);
                }
                currentQuestion = line.Trim();
                answers.Clear();
            }
            else if (line.StartsWith("-"))
            {
                answers.Add(line.Trim());
            }
        }
        if (!string.IsNullOrEmpty(currentQuestion))
        {
            DisplayQuestion(currentQuestion, answers, userResponses);
        }
        return userResponses;
    }

    static void DisplayQuestion(string question, List<string> answers, List<Tuple<string, string>> userResponses)
    {
        Console.WriteLine($"Question: {question}");
        for (int i = 0; i < answers.Count; i++)
        {
            Console.WriteLine($"  {i + 1}. {answers[i]}");
        }
        Console.Write("Select your answer (number): ");
        string response = Console.ReadLine();
        userResponses.Add(new Tuple<string, string>(question, response));
    }

    static void SaveResults(List<Tuple<string, string>> responses, string outputPath)
    {
        using (StreamWriter writer = new StreamWriter(outputPath))
        {
            foreach (var response in responses)
            {
                writer.WriteLine($"{response.Item1} - Answer: {response.Item2}");
            }
        }
    }
}

File: src/cli_tools/SEF_CLI_Tool.py

import argparse
import datetime

def load_exam(file_path):
    with open(file_path, 'r') as file:
        return file.readlines()

def display_exam(questions):
    print("Starting the Exam:")
    user_responses = []
    for question in questions:
        if not question.startswith('-'):
            print(f"Question: {question.strip()}")
            answers = []
        elif question.startswith('-'):
            answers.append(question.strip())
            for idx, answer in enumerate(answers):
                print(f"  {idx + 1}. {answer}")
            response = input("Select your answer (number): ")
            user_responses.append((question.strip(), response))
    return user_responses

def save_results(responses, output_path):
    with open(output_path, 'w') as file:
        for question, response in responses:
            file.write(f"{question} - Answer: {response}\n")

def main():
    parser = argparse.ArgumentParser(description="SEF Exam CLI Tool")
    parser.add_argument('file', help="Path to the SEF formatted exam file")
    parser.add_argument('-o', '--output', help="Path to save the exam results", default="results.txt")
    args = parser.parse_args()

    questions = load_exam(args.file)
    responses = display_exam(questions)
    save_results(responses, args.output)
    print(f"Results saved to {args.output}")

if __name__ == "__main__":
    main()
